pkg/operator/guardbudget/guard_budget_controller.go

```diff
@@ -0,0 +1,195 @@
+		},
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "GuardBudgetController"),
+		eventRecorder: eventRecorder.WithComponentSuffix("guard-budget-controller"),
+	}
+
+	kubeInformersForNamespaces.InformersFor("openshift-machine-config-operator").Policy().V1beta1().PodDisruptionBudgets().Informer().AddEventHandler(c.eventHandler())
+	kubeInformersForNamespaces.InformersFor("openshift-etcd").Policy().V1beta1().PodDisruptionBudgets().Informer().AddEventHandler(c.eventHandler())
+	return c
+}
+
+func (c *GuardBudgetController) sync() error {
+	err := c.checkGuardBudget()
+	if err != nil {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "GuardBudgetDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "Error",
+			Message: err.Error(),
+		}))
+		if updateErr != nil {
+			c.eventRecorder.Warning("GuardBudgetErrorUpdatingStatus", updateErr.Error())
+		}
+		return err
+	}
+
+	_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient,
+		v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:   "GuardBudgetDegraded",
+			Status: operatorv1.ConditionFalse,
+			Reason: "AsExpected",
+		}))
+	return updateErr
+}
+
+func (c *GuardBudgetController) checkGuardBudget() error {
+	// checks if quorum-guard pdb exists in mco namespace
+	machineConfigOperatorPDB, err := c.machineConfigOperatorPodDisruptionBudgetLister.PodDisruptionBudgets("openshift-machine-config-operator").Get("etcd-quorum-guard")
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// removed condition
+			_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+				Type:    conditionBootstrapRemoved,
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "GuardBudgetRemoved",
+				Message: "Quorum guard pod disruption budget has been removed from MCO namespace",
+			}))
+			if updateErr != nil {
+				c.eventRecorder.Warning("GuardBudgetErrorUpdatingStatus", updateErr.Error())
+				return updateErr
+			}
+			// return because no work left to do
+			return nil
+		} else {
+			return err
+		}
+	}
+
+	_, _, _ = v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+		Type:    conditionBootstrapRemoved,
+		Status:  operatorv1.ConditionFalse,
+		Reason:  "GuardBudgetNotRemoved",
+		Message: fmt.Sprintf("Quorum guard pod disruption budget is not removed yet: machineConfigOperatorPDB %#v", machineConfigOperatorPDB),
+	}))
+
+	if err := c.kubeClient.PolicyV1beta1().PodDisruptionBudgets("openshift-machine-config-operator").Delete("etcd-quorum-guard", nil); err != nil {
+		c.eventRecorder.Event("GuardBudgetController", "failed to remove guard budget")
+	}
+
+	_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
```

Comment:
no, you'll get triggered again because of the delete. Let the re-enter on this loop set it.